Dockerfile

```diff
@@ -21,31 +30,41 @@ RUN test -z "$(gofmt -l $(find . -type f -name '*.go' -not -path "./vendor/*"))"
 # ldflags -X injects commit version into binary
 RUN /usr/bin/license-check -path ./ --verbose=false "Alex Ellis" "OpenFaaS Author(s)"
 
-RUN go test $(go list ./... | grep -v /vendor/ | grep -v /template/|grep -v /build/|grep -v /sample/) -cover
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} \
+    go test $(go list ./... | grep -v /vendor/ | grep -v /template/|grep -v /build/|grep -v /sample/) -cover
 
-RUN VERSION=$(git describe --all --exact-match `git rev-parse HEAD` | grep tags | sed 's/tags\///') \
-    && GIT_COMMIT=$(git rev-list -1 HEAD) \
-    && CGO_ENABLED=0 GOOS=linux go build --ldflags "-s -w \
+RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} CGO_ENABLED=0 \
+    go build --ldflags "-s -w \
     -X github.com/openfaas/faas-cli/version.GitCommit=${GIT_COMMIT} \
     -X github.com/openfaas/faas-cli/version.Version=${VERSION} \
-    -X github.com/openfaas/faas-cli/commands.Platform=x86_64" \
+    -X github.com/openfaas/faas-cli/commands.Platform=${TARGETARCH}" \
     -a -installsuffix cgo -o faas-cli
 
 # CICD stage
-FROM alpine:3.12 as root
+FROM --platform=${BUILDPLATFORM:-linux/amd64} alpine:3.12 as root
+
+ARG NS
+ARG REPO
+
+LABEL org.opencontainers.image.source https://github.com/$NS/$REPO
```

Comment:
Does that string need to be split out?

Reply:
No it doesn't need to be split out. I'll put a `=` in between.

Reply:
Can we not inject the complete URL instead?

Reply:
If you mean injecting the whole `org.opencontainers.image.source=https://github.com/$REPO` into the Dockerfile with build args, then I think it's not possible. The Dockerfile expects two args in LABEL, I get a red squiggly line saying so.
<img width="368" alt="Screenshot 2020-11-17 at 1 54 01 PM" src="https://user-images.githubusercontent.com/25264581/99364740-5a6de600-28dc-11eb-9887-123201ab608e.png">

Reply:
Okay I get it, by URL you mean this> https://github.com/openfaas/faas-cli/pull/839#discussion_r524514293